Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -379,91 +379,133 @@ PTF_TEST_CASE(TestPcapLiveDeviceStatsMode)
+		// try to stop capture from within the callback, verify no impact on capturing
+		packetCount = 0;
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStopCapture, &packetCount, 10), 1);
+		PTF_ASSERT_EQUAL(packetCount, 5);
 
-	// verify it's possible to capture blocking mode after non-blocking mode
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeNoTimeout, &packetCount, 30), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+		// verify it's possible to capture non-blocking after the mess done in previous lines
+		packetCount = 0;
+		PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
 
-	// try to start capture from within the callback, verify no error
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStartCapture, &packetCount, 30), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+		// verify an error returns if trying capture blocking while non-blocking is running
+		pcpp::Logger::getInstance().suppressLogs();
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 1), 0);
+		pcpp::Logger::getInstance().enableLogs();
 
-	// try to stop capture from within the callback, verify no impact on capturing
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStopCapture, &packetCount, 10), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+		totalSleepTime = 0;
+		while (totalSleepTime <= 5)
+		{
+			pcpp::multiPlatformSleep(1);
+			totalSleepTime += 1;
+			if (packetCount > 0)
+				break;
+		}
 
-	// verify it's possible to capture non-blocking after the mess done in previous lines
-	packetCount = 0;
-	PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
+		PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
 
-	// verify an error returns if trying capture blocking while non-blocking is running
-	pcpp::Logger::getInstance().suppressLogs();
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 1), 0);
-	pcpp::Logger::getInstance().enableLogs();
+		liveDev->stopCapture();
+		PTF_ASSERT_GREATER_THAN(packetCount, 0);
 
-	totalSleepTime = 0;
-	while (totalSleepTime <= 5)
-	{
-		pcpp::multiPlatformSleep(1);
-		totalSleepTime += 1;
-		if (packetCount > 0)
-			break;
+		liveDev->close();
+
+		// a negative test
+		pcpp::Logger::getInstance().suppressLogs();
+		PTF_ASSERT_FALSE(liveDev->startCapture(packetArrives, &packetCount));
+		pcpp::Logger::getInstance().enableLogs();
 	}
+} // TestPcapLiveDeviceBlockingMode
 
-	PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
+PTF_TEST_CASE(TestPcapLiveDeviceBlockingModePollTimeout)
+{
+#if !defined(_WIN32)
+	// drop all packets on the interface
+	const char* iptablesAddInputDrop = "sudo iptables -A INPUT -i eth0 -j DROP";
+	const char* iptablesAddOutputDrop = "sudo iptables -A OUTPUT -o eth0 -j DROP";
```

Comment:
I just noticed we hard-code the interface `eth0`. This is problematic because the interface name may change per machine. Do you know if it's possible to block traffic with `iptables` using an IP address which is available in `PcapTestGlobalArgs.ipToSendReceivePackets`?

Reply:
yes, it's possible, however, we need to use additional code to find out the interface name of the IP